engine-api/src/main/java/io/nosqlbench/engine/api/activityimpl/SimpleActivity.java

```diff
@@ -652,19 +652,17 @@ protected <O> OpSequence<OpDispenser<? extends O>> createOpSequence(Function<OpT
     protected StmtsDocList loadStmtsDocList() {
 
         try {
-            StmtsDocList stmtsDocList = null;
-
             Optional<String> stmt = activityDef.getParams().getOptionalString("op", "stmt", "statement");
             Optional<String> op_yaml_loc = activityDef.getParams().getOptionalString("yaml", "workload");
             if (stmt.isPresent()) {
-                stmtsDocList = StatementsLoader.loadStmt(logger, stmt.get(), activityDef.getParams());
                 workloadSource = "commandline:" + stmt.get();
+                return StatementsLoader.loadStmt(logger, stmt.get(), activityDef.getParams());
             } else if (op_yaml_loc.isPresent()) {
-                stmtsDocList = StatementsLoader.loadPath(logger, op_yaml_loc.get(), activityDef.getParams(), "activities");
                 workloadSource = "yaml:" + op_yaml_loc.get();
+                return StatementsLoader.loadPath(logger, op_yaml_loc.get(), activityDef.getParams(), "activities");
```

Comment:
On line below, do we need to provide another else block to warn that neither the statement and the op yaml location was present.

Reply:
I left the responsibility of this to the consumer, which has more proactive checks now. There may be cases where we want to allow automation to proceed at this level. It's not clear yet.